syntax = "proto3";

package order;

message Order {
  int32 id = 1;
  int32 user_id = 2;
  int32 product_id = 3;
  int32 quantity = 4;
  double total_price = 5;
}

message CreateOrderRequest {
  int32 user_id = 1;
  int32 product_id = 2;
  int32 quantity = 3;
}

message OrderResponse {
  int32 code = 1;
  string message = 2;
  Order order = 3;
}

message ListOrdersRequest {}

message OrderList {
  repeated Order orders = 1;
}

service OrderService {
  rpc CreateOrder(CreateOrderRequest) returns (OrderResponse);
  rpc ListOrders(ListOrdersRequest) returns (OrderList);
}
